### PR TITLE
[BUG] Fix plot_interval key selection after interval subsetting

### LINKED ISSUE #9523: [BUG] plot_series raises KeyError when plotting interval of multivariate time series
**Describe the bug**

When using `sktime.utils.plotting.plot_series` to plot the predicted interval of a multivariate time series forecasting problem, the code raises `KeyError`.

**To Reproduce**

```python
from sktime.datasets import load_longley
from sktime.utils.plotting import plot_series
from sktime.forecasting.var import VAR
import matplotlib.pyplot as plt


_, y = load_longley()

forecaster_var = VAR()
forecaster_var.fit(y_train, fh=fh)
y_pred_interval = forecaster_var.predict_interval(coverage=0.90)


fig, ax = plt.subplots(figsize=(10, 5))
plot_series(
    y_train["GNPDEFL"],
    y_test["GNPDEFL"],
    pred_interval=y_pred_interval[["GNPDEFL"]],
    labels=["Train", "Test"],
    ax=ax,
    title="GNP Deflator",
)
plt.show()
```

**Expected behavior**
The interval plot.

**Additional context**

The error might be in `plot_interval`, specifically in the line that does:

```python
var_name = interval_df.columns.levels[0][0]
```

Which is not safe, since `levels` keep previous columns even after selecting a subset.

**Versions**
<details>
System:
    python: 3.11.11 (main, Dec 26 2024, 12:31:23) [Clang 16.0.0 (clang-1600.0.26.6)]
executable: [/Users/felipeangelim/Workspace/sktime/.venv/bin/python](https://file+.vscode-resource.vscode-cdn.net/Users/felipeangelim/Workspace/sktime/.venv/bin/python)
   machine: macOS-26.3-arm64-arm-64bit

Python dependencies:
          pip: 25.2
       sktime: 0.39.0
      sklearn: 1.7.2
       skbase: 0.12.6
        numpy: 2.3.5
        scipy: 1.17.0
       pandas: 2.3.3
   matplotlib: 3.10.8
       joblib: 1.5.3
huggingface-hub: None
        numba: None
     pmdarima: None
pytorch-forecasting: None
   skforecast: None
        skpro: None
statsforecast: None
  statsmodels: 0.14.6
 transformers: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
</details>

### PR DESCRIPTION
**Pull request by bernardoferreira-png**
#### Reference Issues/PRs
  Fixes #9523

  #### What does this implement/fix? Explain your changes.
  This fixes a `KeyError` in `plot_series(..., pred_interval=...)` when plotting
  a single variable from a multivariate prediction interval DataFrame.
  The previous code used `interval_df.columns.levels[0][0]`, which may point to
  an unused level after subsetting.
  This PR uses `interval_df.columns.get_level_values(0)[0]` to select the
  actually present variable label.

  #### Does your contribution introduce a new dependency? If yes, which one?
  No.

  #### What should a reviewer concentrate their feedback on?
  - Correctness of level-0 variable selection in `plot_interval`.
  - Regression coverage for non-first-variable subsetting.

  #### Did you add any tests for the change?
  Yes. Updated `test_plotting_dataframe_with_unused_levels` to use `x2` (non-
  first variable), which reproduces the previous failure path and guards against
  regression.

  #### Any other comments?
  Local verification:
  - `pytest sktime/utils/tests/test_plotting.py` -> 46 passed
  - Repro script for issue #9523 no longer raises `KeyError`

### CODE REVIEW

**Comment by felipeangelimvieira** [comment]
@bernardoferreira-png  could you kindly add yourself and @codervinitjangir to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with code badge?

**Comment by bernardoferreira-png** [comment]
Hello @felipeangelimvieira, thanks for your reply. Perhaps I should wait for feedback from @fkiraly since he flagged the PR?

@codervinitjangir, thanks for the review as well.

@fkiraly, would you prefer that I trim or simplify the comments to make them less AI-heavy? I appreciate the candid feedback as this is my first PR. 

Best regards,
Bernardo

**Comment by felipeangelimvieira** [comment]
> Hello @felipeangelimvieira, thanks for your reply. Perhaps I should wait for feedback from @fkiraly since he flagged the PR?
> 
> @codervinitjangir, thanks for the review as well.
> 
> @fkiraly, would you prefer that I trim or simplify the comments to make them less AI-heavy? I appreciate the candid feedback as this is my first PR. 
> 
> Best regards,
> Bernardo


The changes actually look good, the fix seems ok. Franz marked the PR with AI tag because we have a mechanism to detect when PRs are created by agents, not related to the comments in the code.

If the changes on contributors file is done, it is ready to merge.

**Comment by bernardoferreira-png** [comment]
Thanks for the feedback. Implemented as requested.

I updated test_plotting_dataframe_with_unused_levels to keep coverage for both variables by parametrizing: @pytest.mark.parametrize("variable_to_plot", ["x1", "x2"])

So the test now preserves the original x1 path and also keeps regression coverage for the x2 case that previously triggered the KeyError with the old column-level selection logic.

**Comment by codervinitjangir** [comment]
Checked the latest changes. The parametrization of **test_plotting_dataframe_with_unused_levels** correctly covers both x1 and x2 cases as requested. This ensures the fix works for the new case without breaking existing ones. Verified and LGTM!

**Comment by felipeangelimvieira** [comment]
Thank you @bernardoferreira-png and @codervinitjangir !  Bernardo, can you please add your name and  @codervinitjangir to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with code badge?

**Comment by bernardoferreira-png** [comment]
@felipeangelimvieira sorry for the ignorance but how can I add ourselves? Just edit the .all-contributorsrc and PR?

**Comment by felipeangelimvieira** [comment]
@bernardoferreira-png yes, [this file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc), in this PR

**Comment by duast** [comment]
Hello @felipeangelimvieira, just pushed the change for the .all-contributorsrc. Sorry for the delay. 

Regards,
Bernardo
